docs/architecture/adr-008-blocksize-independent-commitment.md

```diff
@@ -0,0 +1,98 @@
+For example, in the following diagram, you see the message starting at the second share and still having the same commitment. I marked the skipped blocks pink to show that both messages have the same starting index.
+
+![Interactive Commitment](./assets/interactive-commitment.png)
+
+If the message starting point index is larger than the row of `minimalBlocksize`, you take the index mod `minimalBlocksize`. The `minimalBlocksize` is shown in green and the index in pink. In both cases, the commitment stays the same.
+
+![Interactive Commitment 2](./assets/interactive-commitment2.png)
+
+## Alternative Approaches
+
+TODO
+
+## Decision
+
+TODO
+
+## Detailed Design
+
+To implement this decision, you need to change [`CreateCommit`](https://github.com/celestiaorg/celestia-app/blob/0c81704939cd743937aac2859f3cb5ae6368f174/x/payment/types/payfordata.go#L112-166).
+In Detail, [`powerOf2MountainRange`](https://github.com/celestiaorg/celestia-app/blob/0c81704939cd743937aac2859f3cb5ae6368f174/x/payment/types/payfordata.go#L142) should take `minimalBlocksize` as an argument instead of `squareSize`.
+
+Following the non-interactive default rules, `minimalBlocksize` can be calculated like this:
+
+```go
+func minimalBlocksize(mLength uint64) uint64 {
+    blocksize := NextHigherPowerOf2(mLength)
+    //Check if message fits with non-interactive-deafult rules
+    if mLength <= (blocksize * (blocksize -1)) {
+        return blocksize
+    } else {
+        return blocksize << 1
+    }
+}
+```
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Negative
+
+1. The amount of subtroots per commitment increases from O(log(n)) to O(sqrt(n)) while n is the number of shares of a message. The worst case for the most amount of shares is depicted in the diagram below - an entire block missing one share.  
+  ![Interactive Commitment 2](./assets/complexity.png)
+
+2. With more subroots the amount of merkle proofs will increase. With deeper subroots the size of the merkle proofs will increase. So instead of having a merkle proof from the `DataRoot` to the `originalSubRoot` you will need the merkle proof from `DataRoot` to `2^k` amount of `miniSubRoots` with `k` being the height difference of `originalSubRoot` and `miniSubRoots`. You can optimize this by having one merkle proof from `DataRoot` to `originalSubRoot` and then `k` merkle proofs from `originalSubRoot` to the `miniSubRoots`. Because the merkle proof is created after the block is published, we know the block size and, therefore, if this `originalSubRoot` exists or not, like in smaller block sizes.
+
+### Positive
+
+1. A Rollup can include the commitment in the block header *before* posting on Celestia because it is size-independent and does not have to wait for Celestia to confirm the block size. In general, the rollup needs access to this commitment in some form to verify a message inclusion proof guaranteeing data availability, which Rollmint currently does not have access to.
```

Comment:
```suggestion
1. A Rollup can include the commitment in the block header *before* posting to Celestia because it is size-independent and does not have to wait for Celestia to confirm the block size. In general, the rollup needs access to this commitment in some form to verify a message inclusion proof guaranteeing data availability, which Rollmint currently does not have access to.
```